zerver/openapi/zulip.yaml

```diff
@@ -6034,6 +6088,216 @@ paths:
                         "result": "error",
                         "msg": "Draft does not exist",
                       }
+  /pinned_views:
+    get:
+      operationId: get-pinned-views
+      tags: ["pinned_views"]
+      summary: Get all pinned views
+      description: |
+        Fetch all pinned views for the current user.
+
+        **Changes**: New in Zulip 10.0 (feature level 324).
+      responses:
+        "200":
+          description: Success.
+          content:
+            application/json:
+              schema:
+                allOf:
+                  - $ref: "#/components/schemas/JsonSuccessBase"
+                  - additionalProperties: false
+                    properties:
+                      result: {}
+                      msg: {}
+                      pinned_views:
+                        type: array
+                        description: |
+                          An array of dictionaries containing the user's pinned views.
+                        items:
+                          $ref: "#/components/schemas/PinnedView"
+                    example:
+                      {
+                        "result": "success",
+                        "msg": "",
+                        "pinned_views":
+                          [
+                            {
+                              "view_id": "inbox",
+                              "location": 2,
+                              "name": null,
+                              "url_hash": null,
+                            },
+                          ],
+                      }
+    post:
+      operationId: add-pinned-view
+      tags: ["pinned_views"]
+      summary: Add a pinned view
+      description: |
+        Add a new pinned view for the current user.
+
+        **Changes**: New in Zulip 10.0 (feature level 324).
+      requestBody:
+        required: true
+        content:
+          application/x-www-form-urlencoded:
+            schema:
+              type: object
+              properties:
+                view_id:
+                  type: string
+                  description: |
+                    The ID of the view to pin.
```

Comment:
This should explain more client is supposed to determine what value to send here.